src/components/radio-group/radio-group.ts

```diff
@@ -0,0 +1,86 @@
+import { html, LitElement } from 'lit';
+import { property, queryAssignedNodes } from 'lit/decorators.js';
+import { IgcRadioComponent } from '../radio/radio';
+import { styles } from './radio-group.css';
+import { watch } from '../common/decorators';
+
+export class IgcRadioGroupComponent extends LitElement {
+  static styles = styles;
+
+  private _cachedRadios!: IgcRadioComponent[];
+
+  @queryAssignedNodes(undefined, true, 'igc-radio')
+  _slottedRadios!: NodeListOf<IgcRadioComponent>;
+
+  private get radios() {
+    return Array.from(this._slottedRadios).filter((radio) => !radio.disabled);
+  }
+
+  private get isLTR(): boolean {
+    const styles = window.getComputedStyle(this);
+    return styles.getPropertyValue('direction') === 'ltr';
+  }
+
+  constructor() {
+    super();
+    this.addEventListener('keydown', this.handleKeydown);
+  }
+
+  @property({ reflect: true, attribute: 'label-position' })
+  labelPosition: 'before' | 'after' = 'after';
+
+  @property({ type: Boolean, reflect: true })
+  disabled = false;
+
+  @property({ reflect: true })
+  alignment: 'vertical' | 'horizontal' = 'vertical';
+
+  @watch('labelPosition', { waitUntilFirstUpdate: true })
+  updateLabelPosition() {
+    this._slottedRadios.forEach(
+      (radio) => (radio.labelPosition = this.labelPosition)
+    );
+  }
+
+  @watch('disabled', { waitUntilFirstUpdate: true })
+  updateDisabled() {
+    if (!this._cachedRadios) this._cachedRadios = this.radios;
+    this._cachedRadios.forEach((radio) => (radio.disabled = this.disabled));
```

Comment:
I understand that you want to modify only the radios that initially are not disabled, but what if radio buttons are added/removed at runtime?
This makes me think what we should listen for `slotchange` and modify the newly added radios' labelPosition and disabled properties. I think that we should either override all radios' properties or remove these properties from the radio group component.

Reply:
I am on the side of overriding the values specified by individual radio buttons. So we should drop all caching that I am currently doing for all radio buttons in a group. Do you think we should do this on first update as well?

Reply:
I am more on the side of removing these props from the radio group :)
@rkaraivanov, What do you think?
If we leave them I think they should definitely work on the first update which may cause overriding the individual radio's values.

Reply:
We have to keep parity with igniteui-angular, this is the only reason they're there.